geode-core/src/test/java/org/apache/geode/distributed/internal/InternalLocatorTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.apache.geode.distributed.internal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.junit.Test;
+
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.internal.HttpService;
+import org.apache.geode.internal.cache.InternalCacheForClientAccess;
+import org.apache.geode.internal.cache.InternalRegionFactory;
+import org.apache.geode.internal.security.SecurableCommunicationChannel;
+import org.apache.geode.logging.internal.LoggingSession;
+import org.apache.geode.management.internal.BaseManagementService;
+
+public class InternalLocatorTest {
+  @Test
+  public void startClusterManagementServiceWithRestServiceEnabledInvokesStartManager()
+      throws IOException {
+    LoggingSession loggingSession = mock(LoggingSession.class);
```

Comment:
I think most of the common setup stuff could be moved into a setup method. Then only the stuff that differs from one test to another would remain. That would make the test-specific details stand out more, and would more clearly express the intent of each test.

If only a few setup details remain in each test, and the intent is obvious, that's enough. If there are more than a few details, perhaps they can be hidden in helper methods with that express the intent.

Reply:
I refactored common code into a couple of helper methods. I hope that at some point this unit test will have more coverage for this class so I did not put this code into a common setup method. I think most of this code is specific to testing the "startClusterManagementService" method.